scriptworker/poll.py

```diff
@@ -121,15 +127,33 @@ def get_azure_urls(context):
 
     Returns:
         dict: the claimTask json
+
+    Raises:
+        ScriptWorkerException: on failure of request_function
```

Comment:
We don't catch a `ScriptWorkerException` in the first `request_function` call.  Also, in `find_task`, we return None and then delete from Azure.  I could take a stab at fixing these docstrings; do you have a recommendation on how to word?